### PR TITLE
push to stable

### PR DESCRIPTION
**Pull request by vanhauser-thc**
before we get a larger readme change from Jana maybe we should push to stable.

### CODE REVIEW

**Review by domenukk — APPROVED** [review]

File: frida_mode/DEBUGGING.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+It should be noted that attempting to debug code using gdb which has been
+instrumented in FRIDA is unlikely to be successful since the debugger will be
+inserting breakpoints by patching the code in memory. FRIDA works by reading
+this code and generating an instrumented copy to execute. In any case, unless
+you are very familiar with the implementation of Stalker, the instrumented code
+generated by FRIDA is likely to be very difficult to follow. For this reason,
+the following debugging strategies are outlined below.
+
+Byte convention below all files should be provided with their path (they are
```
Comment (hexcoder-) [inline frida_mode/DEBUGGING.md]: "By convention" instead of "Byte convention"?

File: instrumentation/README.persistent_mode.md [hunk 2]
```diff
@@ -2,13 +2,16 @@
 
 ## 1) Introduction
 
-The most effective way is to fuzz in persistent mode, as the speed can easily
+In persistent mode, AFL++ fuzzes a target multiple times
+in a single process, instead of forking a new process for each fuzz execution.
```
Comment (hexcoder-) [inline instrumentation/README.persistent_mode.md]: This is technically unprecise, I think, as the forkserver already avoided fork()ing and used clone() and COW. 
The main points here (for me) is: 
 - the life time of the target is extended over multiple fuzzing runs.
 - The target implements a loop and runs like a nonterminating server waiting for inputs.
 - AFL supplies these inputs.

File: instrumentation/SanitizerCoverageLTO.so.cc [hunk 3]
```diff
@@ -626,12 +626,41 @@ bool ModuleSanitizerCoverage::instrumentModule(
...
+                             !FuncName.compare("Curl_safe_strcasecompare") ||
+                             !FuncName.compare("cmsstrcasecmp"));
+            isStrncasecmp &= (!FuncName.compare("strncasecmp") ||
+                              !FuncName.compare("strnicmp") ||
+                              !FuncName.compare("ap_cstr_casecmpn") ||
+                              !FuncName.compare("OPENSSL_strncasecmp") ||
+                              !FuncName.compare("xmlStrncasecmp") ||
+                              !FuncName.compare("g_ascii_strncasecmp") ||
+                              !FuncName.compare("Curl_strncasecompare") ||
+                              !FuncName.compare("g_strncasecmp"));
```
Comment (hexcoder-) [inline instrumentation/SanitizerCoverageLTO.so.cc:662]: If these lists keep on growing, we could soon have namespace issues (clashes with user functions named the same).

**Review by hexcoder- — APPROVED** [review]
LGTM, thanks.

**Review by andreafioraldi — APPROVED** [review]

File: frida_mode/DEBUGGING.md (hunk 1 above)
Comment (vanhauser-thc) [reply frida_mode/DEBUGGING.md]: @WorksButNotTested

File: instrumentation/SanitizerCoverageLTO.so.cc (hunk 3 above)
Comment (vanhauser-thc) [reply instrumentation/SanitizerCoverageLTO.so.cc:662]: that is not an issue IMHO. it is not only matched on function name but also parameter types and return type. and even if these all match and still it does something completely different the impact is minimal

File: instrumentation/README.persistent_mode.md (hunk 2 above)
Comment (vanhauser-thc) [reply instrumentation/README.persistent_mode.md]: I tried to clarify it a little bit better. I am not sure that your main points make it easier to understand either, could work for some and not for others. I think this only works with seeing example code, whatever the writing.

File: frida_mode/DEBUGGING.md (hunk 1 above)
Comment (WorksButNotTested) [reply frida_mode/DEBUGGING.md]: Yep. Typo sorry.

Comment (hexcoder-) [reply frida_mode/DEBUGGING.md]: fixed, Thanks.

File: instrumentation/README.persistent_mode.md (hunk 2 above)
Comment (hexcoder-) [reply instrumentation/README.persistent_mode.md]: Yes, I agree; lets delegate this to an example.

File: instrumentation/SanitizerCoverageLTO.so.cc (hunk 3 above)
Comment (hexcoder-) [reply instrumentation/SanitizerCoverageLTO.so.cc:662]: ok with me. Lets see, if someone experiences this conflict at all.

**Pull request merged**